python3Packages: don't propagate build-platform cffi

lopsided98 opened the pull request:
###### Description of changes

#142301 added `cffi` to `propagatedNativeBuildInputs` in many packages to fix cross-compilation. This had the unfortunate effect of making build-platform `cffi` part of the runtime closure of all downstream packages.

On the other hand, it appears that it is not actually necessary to propagate `cffi` to fix cross-compilation; instead it is enough to add it to `nativeBuildInputs`.

This patch implements this change, while also reverting #200779 (my initial attempt at fixing build-platform contamination in `brotlicffi`) and removing a few other `propagatedNativeBuildInputs` that were unnecessarily added in #142301.

`argon2-cffi` fails for other reasons, so I can't check exactly which dependencies are required.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] armv6l-linux (cross)
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

cc @legendofmiracles @Artturin

lopsided98 committed:
python3Packages: don't propagate build-platform cffi

cross-compilation. This had the unfortunate effect of making build-platform
cffi part of the runtime closure of all downstream packages.

On the other hand, it appears that it is not actually necessary to propagate
cffi to fix cross-compilation; instead it is enough to add it to
nativeBuildInputs.

This patch implements this change, while also reverting #200779 (my initial
attempt at fixing build-platform contamination in brotlicffi) and removing a
few other propagatedNativeBuildInputs that were unnecessarily added in #142301.

lopsided98 commented:
I rebased and fixed the conflicts.

ether42 commented:
> argon2-cffi fails for other reasons, so I can't check exactly which dependencies are required.

Are you perhaps also hitting this (when compiling `argon2-cffi-bindings` though)?
```
armv6l-unknown-linux-gnueabihf-gcc: error: unrecognized command-line option ‘-msse2’ 
```
I was able to work around that by setting: [`ARGON2_CFFI_USE_SSE2 = 0`](https://github.com/hynek/argon2-cffi-bindings/blob/89df7eac3fd31d3d500b97783af9a05961b1564c/src/_argon2_cffi_bindings/_ffi_build.py#L12).

At least, your patch does help since it fixes:
```
ModuleNotFoundError: No module named '_cffi_backend'
```
for me.

pjungkamp committed:
blackbox-terminal: add mainProgram

Atemu committed:
dnscrypt-proxy: rename from dnscrypt-proxy2

An alias is added for dnscrypt-proxy2

eclairevoyant committed:
muse: 3.1.1 -> 4.1.0

veehaitch committed:
github-runner: 2.307.1 -> 2.308.0

* Update runner to v2.308.0
* Use Node 20 by default
* Allow building with Node 16 support through `nodeRuntimes` argument

It's important to have an easy optin for Node 16 since quite many GitHub
actions (explicitly) rely on Node 16. It probably will take some time
until most switch to Node 20. Note that it's also not an option to tell
those who need Node 16 support to continue to use an older version since
GitHub removes support for previous releases quite quickly.

arcnmx committed:
xss-lock: add meta.mainProgram

See #246386

veehaitch committed:
nixos/github-runners: add `nodeRuntimes` option

pinpox committed:
freecad: 0.20.2 -> 0.21.0

jmbaur committed:
coreboot-toolchain: 4.20 -> 4.21

https://doc.coreboot.org/releases/coreboot-4.21-relnotes.html#toolchain

simonrw committed:
shaka-packager: init at 2.6.1

mjm committed:
elixir-ls: meta.mainProgram

hzeller committed:
verible: 0.0.3253 -> 0.0.3410

dritter committed:
josm: 18789 -> 18822

Commit:
zsh-completions: 0.34.0 -> 0.35.0

Ma27 committed:
linux: 4.14.323 -> 4.14.324

Ma27 committed:
linux: 4.19.292 -> 4.19.293

Ma27 committed:
linux: 5.10.192 -> 5.10.193

Ma27 committed:
linux: 5.15.128 -> 5.15.129

Ma27 committed:
linux: 5.4.254 -> 5.4.255

Ma27 committed:
linux: 6.1.49 -> 6.1.50

Ma27 committed:
linux: 6.4.12 -> 6.4.13

Ma27 committed:
linux-rt_5_4: 5.4.248-rt83 -> 5.4.254-rt85

r-ryantm committed:
octave: 8.2.0 -> 8.3.0

AndersonTorres committed:
freecad: 0.21.0 -> 0.21.1

Commit:
emacs: add missing `libXi`  dependency when enabling `withXinput2`

Without libXi, the configure flag `--with-xinput2` has no effect.

This can be verified by looking up the variable `system-configuration-features` in Emacs.

r-ryantm committed:
monkeysAudio: 10.19 -> 10.20

fabaff committed:
python311Packages.peaqevcore: 19.2.0 -> 19.2.1

Changelog: https://github.com/elden1337/peaqev-core/releases/tag/19.2.1

r-ryantm committed:
jfrog-cli: 2.45.0 -> 2.46.2

r-ryantm committed:
timeular: 5.9.0 -> 6.2.2

r-ryantm committed:
nmon: 16n -> 16p

r-ryantm committed:
python310Packages.jaraco-logging: 3.1.2 -> 3.2.0

fabaff committed:
python311Packages.pyinstrument: 4.5.0 -> 4.5.2

Diff: https://github.com/joerick/pyinstrument/compare/refs/tags/v4.5.0...v4.5.2

fabaff committed:
python311Packages.pytapo: 3.1.18 -> 3.2.14

fabaff committed:
python311Packages.pytibber: 0.28.1 -> 0.28.2

Diff: https://github.com/Danielhiversen/pyTibber/compare/refs/tags/0.28.1...0.28.2

Changelog: https://github.com/Danielhiversen/pyTibber/releases/tag/0.28.2

fabaff committed:
python311Packages.requirements-detector: 1.1.0 -> 1.2.2

Diff: https://github.com/landscapeio/requirements-detector/compare/1.1.0...1.2.2

fabaff committed:
python311Packages.rich-argparse: 1.1.0 -> 1.3.0

Diff: https://github.com/hamdanal/rich-argparse/compare/v1.1.0...v1.3.0

Changelog: https://github.com/hamdanal/rich-argparse/blob/v1.3.0/CHANGELOG.md

fabaff committed:
python311Packages.riscv-config: 3.5.2 -> 3.13.1

Diff: https://github.com/riscv-software-src/riscv-config/compare/refs/tags/3.5.2...3.13.1

Changelog: https://github.com/riscv-software-src/riscv-config/blob/3.13.1/CHANGELOG.md

fabaff committed:
python311Packages.requirements-detector: add changelog to meta

r-ryantm committed:
python311Packages.aws-lambda-builders: 1.36.0 -> 1.37.0

r-ryantm committed:
uxn: unstable-2023-08-10 -> unstable-2023-08-30

fabaff committed:
Merge pull request #252735 from fabaff/peaqevcore-bump

python311Packages.peaqevcore: 19.2.0 -> 19.2.1

fabaff committed:
python310Packages.jaraco-logging: add changelog to meta

fabaff committed:
python310Packages.jaraco-logging: disable on unsupported Python releases

- remove six

fabaff committed:
python311Packages.pyinstrument: add changelog to meta

doronbehar committed:
Merge pull request #252736 from r-ryantm/auto-update/monkeysAudio

monkeysAudio: 10.19 -> 10.20

fabaff committed:
python311Packages.pyinstrument: disable on unsupported Python releases

doronbehar committed:
Merge pull request #252629 from r-ryantm/auto-update/octave

octave: 8.2.0 -> 8.3.0

fabaff committed:
python311Packages.pyinstrument: add format

Ma27 committed:
Merge pull request #252605 from Ma27/linux-kernel-updates

Linux kernel updates 2023-08-30

fabaff committed:
Merge pull request #252759 from r-ryantm/auto-update/python311Packages.aws-lambda-builders

python311Packages.aws-lambda-builders: 1.36.0 -> 1.37.0

fabaff committed:
Merge pull request #252758 from fabaff/pytibber-bump

python311Packages.pytibber: 0.28.1 -> 0.28.2

fabaff committed:
Merge pull request #252757 from fabaff/riscv-config-bump

python311Packages.riscv-config: 3.5.2 -> 3.13.1

fabaff committed:
Merge pull request #252756 from fabaff/rich-argparse-bump

python311Packages.rich-argparse: 1.1.0 -> 1.3.0

fabaff committed:
Merge pull request #252755 from fabaff/requirements-detector-bump

python311Packages.requirements-detector: 1.1.0 -> 1.2.2

fabaff committed:
Merge pull request #252749 from fabaff/pytapo-bump

python311Packages.pytapo: 3.1.18 -> 3.2.14

NickCao committed:
Merge pull request #252133 from hzeller/20230829-update-verible

verible: 0.0.3253 -> 0.0.3410

r-ryantm committed:
python310Packages.cmaes: 0.9.1 -> 0.10.0

fabaff committed:
python311Packages.prettytable: 3.5.0 -> 3.8.0

Diff: jazzband/prettytable@refs/tags/3.5.0...3.8.0

Changelog: https://github.com/jazzband/prettytable/releases/tag/3.8.0

Atemu committed:
Merge pull request #248669 from Atemu/dnscrypt-proxy-update-alias

dnscrypt-proxy: rename from dnscrypt-proxy2

fabaff committed:
Merge pull request #252751 from fabaff/pyinstrument-bump

python311Packages.pyinstrument: 4.5.0 -> 4.5.2

fabaff committed:
Merge pull request #252744 from r-ryantm/auto-update/python310Packages.jaraco-logging

python310Packages.jaraco-logging: 3.1.2 -> 3.2.0

fabaff committed:
python310Packages.cmaes: add changelog to meta

matthiasbeyer committed:
Merge pull request #252738 from r-ryantm/auto-update/jfrog-cli

jfrog-cli: 2.45.0 -> 2.46.2

matthiasbeyer committed:
Merge pull request #252740 from r-ryantm/auto-update/timeular

timeular: 5.9.0 -> 6.2.2

matthiasbeyer committed:
Merge pull request #252742 from r-ryantm/auto-update/nmon

nmon: 16n -> 16p

matthiasbeyer committed:
Merge pull request #252539 from Vonfry/update/zsh-completions

zsh-completions: 0.34.0 -> 0.35.0

fabaff committed:
Merge pull request #251813 from fabaff/prettytable-bump

python311Packages.prettytable: 3.5.0 -> 3.8.0

sikmir committed:
Merge pull request #252625 from dritter/update-josm-18822

josm: 18789 -> 18822

zimbatm committed:
Merge pull request #249103 from yaxitech/github-runner-v2.308.0

github-runner: 2.307.1 -> 2.308.0

fabaff committed:
Merge pull request #252774 from r-ryantm/auto-update/python310Packages.cmaes

python310Packages.cmaes: 0.9.1 -> 0.10.0

Enzime committed:
Add `meta.mainProgram` (#252060)

Artturin committed:
Merge pull request #251892 from mjm/push-vpourxpuxwnm

Artturin committed:
Merge pull request #250668 from arcnmx/xss-mainProgram

drupol committed:
Merge pull request #231366 from simonrw/shaka-packager

shaka-packager: init at 2.6.1

Artturin committed:
Merge pull request #249245 from eclairevoyant/muse4

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/why-does-brotlicffi-depend-on-the-wrong-architectures-cffi/44671/2
